Add first python binding test

andrew-catalyst opened the pull request:
Run python tests on CI. Currently just on linux build

andrew-catalyst committed:
Add first python binding test

Run python tests on CI. Currently just on linux build

andrew-catalyst commented on omf-python/tests/test_reader.py:
```diff
@@ -0,0 +1,22 @@
+        self.assertEqual(reader.project.name, "Pyramid")
+        
+    def test_should_raise_expected_file_not_found_exception(self) -> None:
+        # TODO: currently raises pyo3_runtime.PanicException, should raise a sensible exception here
```
Noticed this, we should flesh this out when its fixed.

antbrown commented:
Can ignore `CI / Check (push)` failure in this PR, I've fixed it in PR4.

Is there a quick way I can run the test locally?

andrew-catalyst commented:
> Can ignore `CI / Check (push)` failure in this PR, I've fixed it in PR4.
> 
> Is there a quick way I can run the test locally?

Yes just run `pytest` in the omf-python/ directory ( I added it to requirements.txt )

antbrown commented:
I'll have a go at fixing up the 

```
called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }
``` 

so we can have the `test_should_raise_expected_file_not_found_exception`

Commit:
Split Reader up so it can raise a file io error separately

antbrown commented:
@andrew-catalyst are you able to check the changes I made to implement the file not found test? feel free to change anything I've put in there

antbrown commented:
I didn't know whether to do the test using the `with ...` method, or a `try catch` as this person suggests: https://stackoverflow.com/a/23947350

andrew-catalyst commented:
> I didn't know whether to do the test using the `with ...` method, or a `try catch` as this person suggests: https://stackoverflow.com/a/23947350

That's fine. I usually use `with self.assertRaises(ExceptionType): ...`.

andrew-catalyst committed:
Use assertRaises()

andrew-catalyst commented:
I tweaked the exception test, if that's ok maybe we can merge this

antbrown commented:
> I tweaked the exception test, if that's ok maybe we can merge this

Absolutely, thanks for showing me the better way :+1:
